escrow/contracts/Escrow.cdc

```diff
@@ -0,0 +1,232 @@
+/*
+    Escrow Contract for managing NFTs in a Leaderboard Context.
+    Holds NFTs in Escrow account awaiting transfer or burn.
+
+    Authors:
+        Corey Humeston: corey.humeston@dapperlabs.com
+        Deewai Abdullahi: innocent.abdullahi@dapperlabs.com
+*/
+
+import NonFungibleToken from "./NonFungibleToken.cdc"
+
+pub contract Escrow {
+    // Event emitted when a new leaderboard is created.
+    pub event LeaderboardCreated(name: String, nftType: Type)
+
+    // Event emitted when an NFT is deposited to a leaderboard.
+    pub event EntryDeposited(leaderboardName: String, nftID: UInt64, owner: Address)
+
+    // Event emitted when an NFT is withdrawn from a leaderboard.
+    pub event EntryWithdrawn(leaderboardName: String, nftID: UInt64, owner: Address)
+
+    // Event emitted when an NFT is burned from a leaderboard.
+    pub event EntryBurned(leaderboardName: String, nftID: UInt64)
+
+    // Named Paths
+    pub let CollectionStoragePath: StoragePath
+    pub let CollectionPublicPath: PublicPath
+    pub let CollectionPrivatePath: PrivatePath
+
+    pub struct LeaderboardInfo {
+        pub let name: String
+        pub let nftType: Type
+        pub let entriesLength: Int
+
+        init(name: String, nftType: Type, entriesLength: Int) {
+            self.name = name
+            self.nftType = nftType
+            self.entriesLength = entriesLength
+        }
+    }
+
+    // The resource representing a leaderboard.
+    pub resource Leaderboard {
+        pub var entries: @{UInt64: LeaderboardEntry}
+        pub let name: String
+        pub let nftType: Type
+        pub var entriesLength: Int
+
+        // Adds an NFT entry to the leaderboard.
+        pub fun addEntry(nft: @NonFungibleToken.NFT, leaderboardName: String, depositCap: Capability<&{NonFungibleToken.CollectionPublic}>) {
+            pre {
+                 nft.isInstance(self.nftType): "NFT cannot be used for this leaderboard!"
+            }
+
+            let nftID = nft.id
+
+            // Check if the entry already exists
+            if self.entries[nftID] != nil {
+                panic("Entry already exists for this NFT in the leaderboard")
+            }
+
+            // Create the entry and add it to the entries map
+            let entry <- create LeaderboardEntry(
+                nftID: nftID,
+                ownerAddress: depositCap.address,
+                nft: <-nft,
+                depositCap: depositCap
+            )
+
+            self.entries[nftID] <-! entry
+
+            // Increment entries length.
+            self.entriesLength = self.entriesLength + 1
+
+            emit EntryDeposited(leaderboardName: leaderboardName, nftID: nftID, owner: depositCap.address)
+        }
+
```

Comment:
Might be helpful to document the withdraw and burn functions, explaining their intended use.